### PR TITLE
docs: add annotation for psalm

### PR DESCRIPTION
**Pull request by kenjis**
To reduce psalm issues.

Before: 803 other issues found.
After: ~~684~~ 681 other issues found.

### CODE REVIEW

File: src/Models/UserModel.php [hunk 1]
```diff
@@ -14,6 +14,9 @@
 use CodeIgniter\Shield\Exceptions\ValidationException;
 use Faker\Generator;
 
+/**
+ * @psalm-consistent-constructor
```
Comment (MGatner) [inline src/Models/UserModel.php:18]: Might as well tell PHPStan as well:
```suggestion
 * @phpstan-consistent-constructor
 * @psalm-consistent-constructor
```

https://phpstan.org/blog/solving-phpstan-error-unsafe-usage-of-new-static#add-%40phpstan-consistent-constructor-to-the-class

**Review by MGatner — CHANGES_REQUESTED** [review]

File: src/Models/UserModel.php (hunk 1 above)
Comment (kenjis) [reply src/Models/UserModel.php]: It is not true to all tags, but PHPStan vendor tags are often supported by psalm as well.
So I changed the tag to PHPStan's.

I don't know why but the issues are reduced to 681.

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
Something went wrong with this PR.

I hit the merge button of this PR on GitHub, but I got a form to select my email address.
When I reloaded the PR page, it showed a conflict, so I rebased and force pushed it.
Then the PR was closed., and this PR was messed up.

But when I see the commit log of `develop`, this PR was merged.

**Comment by MGatner** [comment]
Very odd! I've had something similar happen when I accidentally fork another PR off a feature branch, and if the second PR gets merged and then feature is rebased it can "go away". I'm not sure that is what happened here though looking at the weird vomit history of this PR. 

End result appears good, so 🤷‍♂️

**Comment by paulbalandan** [comment]
If this PR was used as base for another PR that got merged first, then this PR is deemed automatically merged. Just read it recently in Github docs.

**Comment by kenjis** [comment]
This is not used as base for another PR.

It seems this PR was merged into develop, but I rebased this branch (with the develop already merged this) and force pushed. The last commit in this PR is `Merge pull request codeigniter4#518 from kenjis/docs-remove-psalm-issues`. `#518` is this PR.

However, I'm still not sure what happened, but as a result, the develop branch looks fine. 
Only this PR page is a mess though.

**Pull request closed without merging**